site/src/pages/UsersPage/UsersPage.tsx

```diff
@@ -91,6 +99,11 @@ export const UsersPage: React.FC = () => {
         isLoading={isLoading}
         canEditUsers={canEditUsers}
         canCreateUser={canCreateUser}
+        filter={usersState.context.filter}
+        onFilter={(query) => {
+          searchParams.set("filter", query)
+          setSearchParams(searchParams)
```

Comment:
Do we actually need both of these? I feel like they're doing the same thing.

Reply:
`.set` is called on the `URLSearchParams` type state object `searchParams`. `setSearchParams` is updating the state with the latest version of `searchParams` after `set` is called.

Reply:
TIL! This looks like a promising simplified approach.